### ISSUE TITLE
Error "require is not defined" when trying to compile to systemjs modules.

### ISSUE THREAD

**Issue by pietschy** [frankwallis/plugin-typescript#47]
Hi there,

I've just updated to typescript 1.6.2 from 1.5.something and am now getting errors which appear to be module related. I also uninstalled and reinstalled the typescript plugin (and am currently using 2.0.17).

```
err  ReferenceError: file:///Users/andrew/projects/concertus/src/main.ts!transpiled:2
     require('angular-material/angular-material.css!');
^
     require is not defined
	Evaluating file:///Users/andrew/projects/concertus/src/main.ts
```

My typescript options are:
```
  typescriptOptions: {
    "noImplicitAny": true,
    "typeCheck": true,
    "resolveAmbientRefs": true,
    "module": "system",
    "target": "ES5",
    "declaration": true,
    "noExternalResolve": false,
    "experimentalDecorators": true,
    "emitDecoratorMetadata": true
  },
```

Not sure if I'm missing something obvious. Any help would be greatly appreciated.

Thanks

**Comment by frankwallis** [comment]
I think this is actually due to a SystemJS change. The fix is to change the ```require``` to an ```import``` so try

```
import('angular-material/angular-material.css!');
```

**Comment by pietschy** [comment]
Howdy, thanks for getting back to me.

My source code has `import 'angular-material/angular-material.css!';` (and used to compile).  From the look of the error message I'd figured it was the transpiler output was the issue (i.e. .../main.ts!transpiled:2). I haven't tried using tsc directly be can give that a go tomorrow and see how I go.

Thanks & cheers

**Comment by pietschy** [comment]
Ok, I've had a bit of a play with this.  Using tsc from the command line with `--module system` works as expected.

I then put some debug lines in transplier.js to see what's happening and it appears ts.createProgram is being called with the correct parameters, but it's generating code with require imports rather than systemjs imports.

My source file is:
```
/// <reference path="../typings/angularjs/angular.d.ts" />
import 'angular-material/angular-material.css!';
import './index.css!';
import {module} from "angular";
import 'angular-material';
import {MainController} from './mainController.class';

let Module = module("main", [
    'ngMaterial'
]);

Module.controller('MainController', MainController);

export default Module;
```

When debugging the compiler options at transpiler.js line:23 is see. 
```
{ noImplicitAny: true,
  typeCheck: true,
  resolveAmbientRefs: true,
  module: 'system',
  target: 1,
  declaration: false,
  noExternalResolve: false,
  experimentalDecorators: true,
  emitDecoratorMetadata: true,
  jsx: 0,
  allowNonTsExtensions: true,
  noResolve: true,
  noLib: true,
  inlineSourceMap: false,
  sourceMap: true,
  isolatedModules: true }
```

And debugging the jstext output after the ts.createProgram(..) gives me:
```
/// <reference path="../typings/angularjs/angular.d.ts" />
require('angular-material/angular-material.css!');
require('./index.css!');
var angular_1 = require("angular");
require('angular-material'); // for some reason import {...} from 'angular-material' doesn't work.
var mainController_class_1 = require('./mainController.class');
var Module = angular_1.module("main", [
    'ngMaterial'
]);
Module.controller('MainController', mainController_class_1.MainController);
exports.default = Module;
```

So it looks like it might be a typescript compiler issue.  I started having a look into the compiler but it was going to take me ages to come up to speed.

**Comment by frankwallis** [comment]
I would try stripping out your config options until it works, I am suspicious of the module="system" setting, it should be a number (the enum value of system). I will have another look later on tonight.

**Comment by frankwallis** [comment]
It seems to me that the built-in typescript compiler may be getting used to transpile this file, it doesn't handle passing the enum compiler options as strings. Have you added the ```packages``` configuration which is needed when using SystemJS 0.16 onwards?

```
  packages: {
    "src": {
      "main": "index",
      "defaultExtension": "ts",
      "meta": {
        "*.ts": {
          "loader": "ts"
        },
        "*.css": {
          "loader": "css"
        }
      }
    }
  },
```

I tried adding angular-material to the example project and it worked fine, so maybe start with that configuration and then try adding in things that you need.

**Comment by pietschy** [comment]
Thanks @frankwallis, turns out I'd left the transpiler set to traceur, changing it to typescript fixed it. 

As a side note, leaving it as traceur and setting my config to use `module: 4` also works - so I'm not sure if that's a bug I should report somewhere.

Thanks for your help.

**Comment by frankwallis** [comment]
Thanks, I have recreated it now and added a fix in 2.1.2

**Comment by pietschy** [comment]
Cool, thanks @frankwallis .  I've installed and it all works again now. Cheers
